Show platform specific shortcut while editing.

Carreau opened the pull request:
As the text in the text field is used to bind the shortcut later introduce a utility function to parse back from platform specific to canonical representation

Closes #5047

## Type of change
<!-- Please delete options that are not relevant. -->
- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update


# How has this been tested?
<!-- Please describe the tests that you ran to verify your changes. -->
- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  


Manually on Macos, checking that I can set shortcuts to things like Ctrl-Meta-Alt-Shift-N
I'm not sure extra tests are needed here. Maybe they will if we find edge cases.

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

goanpeca commented on napari/utils/interactions.py:
```diff
@@ -282,6 +282,28 @@ def __init__(self, shortcut: str):
                 )
 
+    @staticmethod
+    def parse_patflorm(text: str) -> str:
```
```suggestion
    def parse_platform(text: str) -> str:
```

goanpeca commented on napari/_qt/widgets/qt_keyboard_settings.py:
```diff
@@ -379,7 +379,7 @@ def _set_keybinding(self, row, col):
             # get the current item from shortcuts column
             current_item = self._table.currentItem()
-            new_shortcut = current_item.text()
+            new_shortcut = Shortcut.parse_patflorm(current_item.text())
```
```suggestion
            new_shortcut = Shortcut.parse_platform(current_item.text())
```

goanpeca reviewed (dismissed):
Looks good @Carreau, thanks for working on this.

Left a couple of suggestions to fix a typo

Carreau committed:
Show platform specific shortcut while editing.

As the text in the text field is used to bind the shortcut later
introduce a utility function to parse back from platform specific
to canonical representation

Closes #5047

jni reviewed (approved):
I'm approving some more for good measure 😃

aganders3 commented on napari/utils/interactions.py:
```diff
@@ -309,6 +309,28 @@ def __init__(self, shortcut: KeyBindingLike) -> None:
+            text.replace('+', '')
+            text.replace('Plus', '+')
+        for k, v in KEY_SYMBOLS.items():
+            text = text.replace(v, k + '-')
```
@psobolewskiPhD I can reproduce your issue.
I'm not sure this is the _right_ fix, but it's one way to fix your issue. Basically this was trying to bind `"Space-"` which is invalid, so here I only append the extra hyphen if the keybind is longer than the symbol.

```suggestion
        for k, v in KEY_SYMBOLS.items():
            if len(text) > len(v):
                text = text.replace(v, k + '-')
            else:
                text = text.replace(v, k)
```

Carreau replied:
I went for a different fix, because in this case of `⌘⌥` that would not have word, and returned `Meta-Alt-` which also would be invalid.

psobolewskiPhD replied:
Ok, that explains why it's things like space, backspace, etc. that trigger the bug!
I'm not sure what the hyphen is for at all—I never see it—and with this PR I never see the `+` joining keys either, like e.g. Command+M on main.

Previous to the app-model keybindings it looked like the settings.yaml had `-` as the join. 
Currently, I see `+` in my settings.yaml
Still, multi key bindings work with this PR, e.g. `Shift+Meta+Z` (the `+` don't show while editing, which seems like a macOS specific setting.)

I think the `-` might be a leftover and maybe just needs to be removed.

aganders3 replied:
Ah, good point - I just saw your change and it looks good to me!

Carreau replied:
`-` is at least still present in  test , and it is still likely used in extensions I guess. 

I'm playing around with NewType to see if we can avoid using 'Str' everywhere. 

https://github.com/Carreau/napari/pull/new/ntt
